urllib3/response.py

```diff
@@ -351,3 +363,71 @@ def readinto(self, b):
         else:
             b[:len(temp)] = temp
             return len(temp)
+
+    def _read_chunked(self, amt=None):
+        assert self.chunked
+        chunk_left = self.chunk_left
+        value = []
+        while True:
+            if chunk_left is None:
+                line = self._fp.fp.readline(amt)
+                i = line.find(';')
+                if i >= 0:
+                    line = line[:i] # strip chunk-extensions
+                try:
+                    chunk_left = int(line, 16)
+                except ValueError:
+                    # close the connection as protocol synchronisation is
+                    # probably lost
+                    self.close()
+                    raise httplib.IncompleteRead(''.join(value))
+                if chunk_left == 0:
+                    break
+            if amt is None:
+                yield self._fp._safe_read(chunk_left)
+            elif amt < chunk_left:
+                value.append(self._fp._safe_read(amt))
+                self.chunk_left = chunk_left - amt
+                yield ''.join(value)
+            elif amt == chunk_left:
+                value.append(self._fp._safe_read(amt))
+                self._fp._safe_read(2)  # toss the CRLF at the end of the chunk
+                self.chunk_left = None
+                yield ''.join(value)
+            else:
+                yield self._fp._safe_read(chunk_left)
+                amt -= chunk_left
+
+            # we read the whole chunk, get another
+            self._fp._safe_read(2)      # toss the CRLF at the end of the chunk
+            chunk_left = None
+
+        # read and discard trailer up to the CRLF terminator
+        ### note: we shouldn't have any trailers!
+        while True:
+            line = self._fp.fp.readline(amt)
+            if not line:
+                # a vanishingly small number of sites EOF without
+                # sending the trailer
+                break
+            if line == '\r\n':
+                break
+
+        # we read everything; close the "file"
+        self.close()
+
+        yield ''.join(value)
+
+
+    def __iter__(self):
+        """
+        credit/resources:
+         * https://gist.github.com/ericflo/514507
+         * http://blog.dowski.com/2008/04/02/reading-chunked-http11-responses/
+         * https://hg.python.org/cpython/file/2.7/Lib/httplib.py
+        """
+        if not self.chunked:
+            return stream()
```

Comment:
There is. It should probably be iterating over `stream()`, which itself returns a generator. Ideally, we'd use `yield from` here, but sadly Python 2.